transactions.md

```diff
@@ -12,90 +12,169 @@ CockroachDB supports bundling multiple SQL statements into a single all-or-nothi
+> COMMIT;
+~~~
+
+To alert you of failures due to contention, CockroachDB surfaces an error with the code `40001` that begins with the string `retry transaction`.
+
+To handle these types of errors you have two options:
+
+- *Recommended*: Use the `cockroach_restart` functions to create retryable transactions. Retryable transaction can improve performance because their priority's increased each time they are retried, making them more likely to succeed the longer they're in your system. 
+
+  For more information, see [Client-Side Transaction Retries](#client-side-transaction-retries).
+
+- Abort the transaction using `ROLLBACK`, and then reissue all of the statements in the transaction. This does *not* automatically increase the transaction's priority, so it's possible in high-contention workloads for transactions to take an incredibly long time to succeed.
+
+#### Client-Side Transaction Retries
+
+To improve the performance of transactions, CockroachDB include a set of statements that let you retry transaction that fail due to contention errors. Retrying transactions has the benefit of increasing their priority each time they're retried, increasing their likelihood to succeed.
+
+Implementing client-side retries requires three statements:
+
+- [`SAVEPOINT cockroach_restart`](savepoint.html) defines your intent to retry the transaction if there are contention errors. It must be executed after `BEGIN` but before the first statement that manipulates a database.
+- [`ROLLBACK TO cockroach_restart`](rollback-transaction.html#retry-a-transaction) is used when your application detects `40001` / `retry transaction` errors. It provides you a chance to "retry" the transaction by rolling the database's state back to the beginning of the transaction and incrementing the transaction's priority. 
+  
+  After issuing `ROLLBACK TO cockroach_restart`, each of the transaction's statements must then be executed again.
+
+- [`RELEASE SAVEPOINT cockroach_restart`](release-savepoint.html) commits the transaction.
+  
+  You must also execute `COMMIT` afterward to clear the connection for the next transaction.
+
+You can find examples of this in the [Syntax](#syntax) section of this page or at our page [Build a Test App: Execute transaction from a client](build-a-test-app.html#step-4-execute-transactions-from-a-client).
+
+{{site.data.alerts.callout_success}}For Golang users, client-side retries are built into the <code>crdb</code> package of the CockroachDB Go client and happen automatically. For more information, see the <strong>Go</strong> tab of <a href="build-a-test-app.html#step-4-execute-transactions-from-a-client">Build a Test App: Execute transaction from a client</a>.{{site.data.alerts.end}}
```

Comment:
Go, not Golang. We also have this functionality for the SQLAlchemy package in python.